src/RenderView.tsx

```diff
@@ -22,6 +22,40 @@ interface Props {
 export default function RenderView(props: Props) {
 	const width = 800;
 	const height = 450;
+	
+	const forceMeshesComp = useMemo(() => {
+		const mesh1 = new THREE.Mesh(new THREE.ConeGeometry(.02,.2,8),new THREE.MeshBasicMaterial());
+		const mesh2 = new THREE.Mesh(new THREE.ConeGeometry(.02,.2,8),new THREE.MeshBasicMaterial());
+		mesh1.visible = false;
+		mesh2.visible = false;
+		return [mesh1,mesh2];
+	 }, []);
+	
+	/* Position force velocity for the given frame (in props) */
+	 useEffect(() => {
+		const frameData = props.forceData.frames[props.frame]; //select a single frame
+		if (!frameData) return; //animation doesn't depend on forceFileData, so this might be empty from initialization in App.tsx
+		forceMeshesComp.forEach((mesh2,idx) => { //for each un-positioned 3D mesh
+		   const pos1 = frameData.forces[idx].components;
+		   const pos = frameData.forces[idx].position;
+		   if (!pos||isNaN(pos.x)||isNaN(pos.y)||isNaN(pos.z)||(pos.x===0&&pos.y===0&&pos.z===0)) {
+			  mesh2.visible = false; //hide forces with no data
+			  return;
+		   }
+		   var i = 0;
+		   while(i < 600) {
+		   
+		   const compp = pos1.y;
+		   var j = i*.001;
+		   if (i < compp && i+10 > compp) {
```

Comment:
We could use your compp variable directly to set mesh2.position.y, which would eliminate the while loop on line 46 and this conditional which checks if the loop counter is within 10 of the y component value. You could just scale it with the .001 factor on assignment, like so:
```typescript
//line 48
const yComp = pos1.y * .001;
//then, on line 52...
mesh2.position.y = pos.y + yComp;
```